File: app/src/main/java/com/dev/moviedb/mvvm/model/movies/Identifiable.java
package com.dev.moviedb.mvvm.model.movies;

/**
 * An {@link Identifiable} type must be able to give an
 * id that identifies itself;
 *
 * @version 1.0.0
 */
public interface Identifiable {

    /**
     * Return an id that uniquely identifies the instance.
     * @return
     */
    long resolveID();
}

File: app/src/main/java/com/dev/moviedb/mvvm/modelstate/movies/content/Identifiable.kt
package com.dev.moviedb.mvvm.modelstate.movies.content;

/**
 * An {@link Identifiable} type must be able to give an
 * id that identifies itself;
 *
 * Yamda 1.0.0
 */
interface Identifiable {

    /**
     * Return an id that uniquely identifies the instance.
     * @return
     */
    fun resolveID(): Long
}
